<?xml version="1.0"?>
<launch>

  <arg name="real_machine" default="false" />
  <arg name="simulation" default="false" />
  <arg name="robot_ns" default="hydrus" />

  <group ns="$(arg robot_ns)">

    <group if="$(arg real_machine)">
      <group unless="$(arg simulation)">
        <!-- fc & IMU & GPS -->
        <include file="$(find spinal_ros_bridge)/launch/serial.launch" >
          <arg name="baud" value="921600" />
          <arg name="port" value="/dev/flight_controller" />
        </include>

        <!-- mocap -->
        <include file="$(find aerial_robot_base)/launch/external_module/mocap.launch" />

        <!-- leddar one -->
        <include file="$(find leddar_one)/launch/leddar_one.launch" >
          <arg name="port" value="/dev/leddarone" />
        </include>

      </group>
    </group>

    <!-- basic configuration for sensors (e.g. noise sigma) -->
    <rosparam file="$(find hydrus)/config/sensors/imu/spinal.yaml" command="load" />
    <rosparam file="$(find aerial_robot_base)/config/sensors/gps/ublox_m8n.yaml" command="load" />
    <rosparam file="$(find aerial_robot_base)/config/sensors/mocap.yaml" command="load" />
    <rosparam file="$(find aerial_robot_base)/config/sensors/altmeter/leddar_one.yaml" command="load" />

    <!-- <node pkg='hydrus' type='neuron_odom2jointstate.py' name='odom2jointstate' output='screen'> -->
    <!--   <param name="mode" value="rot"/> -->
    <!--   <param name="robot_name" value="$(arg robot_ns)"/> -->
    <!--   <param name='joint_state_topic_name' value='link_torsion/joint_states'/> -->
    <!-- </node> -->

  </group>

</launch>
